src/composables/sdk.ts

```diff
@@ -0,0 +1,66 @@
+import { ref } from 'vue';
+import type {
+  IDefaultComposableOptions,
+  Protocol,
+} from '@/types';
+import { BaseProtocol } from '@/protocols/BaseProtocol';
+import { AeternityProtocol } from '@/protocols/aeternity/AeternityProtocol';
+import { BitcoinProtocol } from '@/protocols/bitcoin/BitcoinProtocol';
+import { DogecoinProtocol } from '@/protocols/dogecoin/DogecoinProtocol';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+  PROTOCOL_DOGECOIN,
+} from '@/popup/utils';
+import { watchUntilTruthy } from '@/utils';
+import { useAccounts } from './accounts';
+
+const isSdkRegistryReady = ref(true);
+const sdkRegistry: Partial<Record<Protocol, BaseProtocol>> = {};
+
+export function useSdk({ store }: IDefaultComposableOptions) {
+  const { activeAccount } = useAccounts({ store });
+
+  function registerSdk(protocol: Protocol, instance: BaseProtocol) {
+    if (sdkRegistry[protocol]) {
+      throw new Error('Protocol instance already exists in the registry');
+    }
+    sdkRegistry[protocol] = instance;
+  }
+
+  function getSdkFromRegistry(protocol: Protocol): BaseProtocol | undefined {
+    return sdkRegistry[protocol];
+  }
+
+  function initSdk(protocol: Protocol): BaseProtocol {
```

Comment:
This is not initiating the SDK, but the protocol class so I think the name is wrong.